core/trino-parser/src/main/java/io/trino/sql/tree/SymbolReference.java

```diff
@@ -63,6 +63,6 @@ public boolean equals(Object o)
     @Override
     public int hashCode()
     {
-        return Objects.hash(name);
+        return name.hashCode();
```

Comment:
The updated code is better, but the reasoning in the commit message is not. Looping over single-element arrays is not a problem. The VM will very likely unroll the loop and eliminate it.

Here's a benchmark result showing the (lack of) performance difference:

```
Benchmark             Mode  Cnt  Score   Error  Units
BenchmarkLoop.direct  avgt   30  2.205 ± 0.019  ns/op
BenchmarkLoop.loop    avgt   30  2.277 ± 0.086  ns/op
```

Reply:
Actually, I removed this change, as it was causing slight differences in generated plans. These changes where non-significant, e.g. order of values in the list, but would still require adapting some tests.